qiskit/pulse/__init__.py

```diff
@@ -87,7 +87,6 @@
    Instruction
 
 .. autosummary::
-   :hidden:
```

Comment:
I don't want these to show up on this docs page! They were added because otherwise their rst pages wouldn't be autogenerated and I couldn't link them, for example:
```
Channels (:mod:`~qiskit.pulse.channels`)
========================================
```
^^^ that wouldn't work. Is this fixed by the new file `docs/apidocs/terra.rst`? If so, all 6 of these lines should be deleted.

Reply:
Unfortunately hidden isn't actually an argument for `autosummary`.
```
qiskit/pulse/__init__.py:docstring of qiskit.pulse:74: WARNING: Error in "autosummary" directive:
unknown option: "hidden".

.. autosummary::
   :hidden:
   :toctree: ../stubs/

   qiskit.pulse.commands
   qiskit.pulse.channels
```

It makes the whole toc added here ignored. It's just an odd coincidence that breaking the autosummary call has the desired effect of rendering the elements but hiding them from the menu structure which is what hidden does for a normal `toctree`.

The terra.rst file is just fixing an oversight from #3894 which forgot to git add the new file after renaming it (right now the docs on master do not have a top level api toc). It won't make any changes for this. The best solution here is to make this a manual toctree with `hidden` and explicitly autoclass the classes instead of relying on autosummary to do this for us.

Reply:
Can we adjust this in a follow up? The change won't be reflected in the published docs until the next release and this PR is big and likely to be broken by another PR touching a docstring so I'd like to get this merged so we don't have to constantly rebase or fix new warnings.